src/test/java/com/cloudant/tests/CloudantClientTests.java

```diff
@@ -208,4 +223,109 @@ public void existingDatabaseCreateException() {
         }
     }
 
+    /**
+     * Check that the connection timeout throws a SocketTimeoutException when it can't connect
+     * within the timeout.
+     */
+    @Test(expected=SocketTimeoutException.class)
+    public void connectionTimeout() throws Throwable {
+
+        //create a socket on port 54321 with only 1 backlog
+        ServerSocket serverSocket = new ServerSocket(54321, 1);
+        //block the socket
+        Socket socket = new Socket();
+        socket.connect(serverSocket.getLocalSocketAddress());
+
+        try {
+            CloudantClient c = new CloudantClient("http://localhost:54321", "", "", new
+                    ConnectOptions().setConnectionTimeout(new TimeoutCustomizationInterceptor
+                    .TimeoutOption(100, TimeUnit.MILLISECONDS)));
+
+            c.createDB("test");
+        } catch (CouchDbException e) {
+            //unwrap the CouchDbException
+            if (e.getCause() != null) {
+                //whilst it would be really nice to actually assert that this was a connect
+                //exception and not some other SocketTimeoutException there are JVM differences in
+                //this respect (i.e. OpenJDK does not appear to distinguish between read/connect)
+                //in its exception messages
+                throw e.getCause();
+            } else {
+                throw e;
+            }
+        } finally {
+            //make sure we close the sockets
+            IOUtils.closeQuietly(serverSocket);
+            IOUtils.closeQuietly(socket);
+        }
+    }
+
+    /**
+     * Checks that the read timeout works. The test sets a read timeout of 0.25 s and the mock
+     * server waits 0.5 s before continuing to send data down the stream so the timeout should be
+     * triggered.
+     */
+    @Test(expected=SocketTimeoutException.class)
+    public void readTimeout() throws Throwable {

```

Comment:
As `SocketTimeoutException` can be thrown for various timeout reasons, I wonder if we should add a couple of additional tests very similar to `connectionTimeout()` and `readTimeout()`, which instead check the converse of these - i.e. we don't get a `SocketTimeoutException` if the connect/reads happen within the timeout. This would give a little extra confidence we're getting the `SocketTimeoutException`s for the reasons we expect to get them in these tests.